### ISSUE TITLE
Fix Bazel incompatible changes

### ISSUE THREAD

**Issue by laurentlb** [bazelbuild/rules_typescript#342]
I'd like to be able to build rules_rust with --all_incompatible_changes:
  `cd rules_typescript && bazel build --all_incompatible_changes //...`

It may not be possible to fix every issue now due to transitive dependencies, but we should try to fix the issues from this repository.

**Comment by manekinekko** [comment]
@laurentlb can we close this issue now that #343 has been merged?

**Comment by laurentlb** [comment]
No. In the PR description, you can see that 5 flags are disabled. The goal is to remove them.

If you want to help, look at these 3 flags:

* `--incompatible_disallow_old_style_args_add` fails due to https://github.com/bazelbuild/rules_typescript/blob/master/internal/protobufjs/ts_proto_library.bzl#L26 (args.add)
* `--incompatible_depset_union=true` fails due to https://github.com/bazelbuild/rules_typescript/blob/master/internal/common/compilation.bzl#L260
* `--incompatible_depset_is_not_iterable=false` fails due to `[f.path for f in type_blacklisted_declarations]` in https://github.com/bazelbuild/rules_typescript/blob/master/internal/common/tsconfig.bzl#L144

For more details about the flags, see https://docs.bazel.build/versions/master/skylark/backward-compatibility.html

**Comment by qzmfranklin** [comment]
Any updates on this?

With version `0.22.1`, I still got errors like this:
```text
ERROR: /home/zhongming/git/LogiOcean/src/protos/BUILD:75:1: in ts_proto_library rule //src/protos:version_def.pb.ts: 
Traceback (most recent call last):
        File "/home/zhongming/git/LogiOcean/src/protos/BUILD", line 75
                ts_proto_library(name = 'version_def.pb.ts')
        File "/home/zhongming/.cache/bazel/_bazel_zhongming/2aea414f4d422ad78870bb37da8563ec/external/build_bazel_rules_typescript/internal/protobufjs/ts_proto_library.bzl", line 81, in _ts_proto_library
                _run_pbjs(ctx.actions, ctx.executable, output_..., <2 more arguments>)
        File "/home/zhongming/.cache/bazel/_bazel_zhongming/2aea414f4d422ad78870bb37da8563ec/external/build_bazel_rules_typescript/internal/protobufjs/ts_proto_library.bzl", line 30, in _run_pbjs
                args.add([f.path for f in proto_files])
        File "/home/zhongming/.cache/bazel/_bazel_zhongming/2aea414f4d422ad78870bb37da8563ec/external/build_bazel_rules_typescript/internal/protobufjs/ts_proto_library.bzl", line 30, in args.add
                [f.path for f in proto_files]
type 'depset' is not iterable. Use the `to_list()` method to get a list. Use --incompatible_depset_is_not_iterable=false to temporarily disable this check.
```

I thought it was already fixed by #378. Just want to raise this here to draw awareness.

**Comment by laurentlb** [comment]
There are still some issues with `--incompatible_depset_is_not_iterable` and `--incompatible_disable_deprecated_attr_params`.
